src/app/src/main/kotlin/org/imperial/mrc/hint/db/CalibrateDataRepository.kt

```diff
@@ -51,11 +58,50 @@ class JooqCalibrateDataRepository: CalibrateDataRepository
         }
     }
 
-    private fun getDataFromConnection(conn: Connection): List<CalibrateResultRow> {
-        val query = DEFAULT_QUERY
+    private fun convertIndicatorToList(resultSet: ResultSet): List<String> {
+        resultSet.use {
+            return generateSequence {
+                if (it.next()) {
+                    it.getString("indicator")
+                } else {
+                    null
+                }
+            }.toList()
+        }
+    }
+
+    private fun getIndicatorQuery(indicator: String): String {
+        return """SELECT
+        age_group,area_id,
+        calendar_quarter,
+        indicator,
+        ROUND(lower, 4) AS lower,
+        ROUND(mean, 4) AS mean,
+        ROUND(mode, 4) AS mode,
+        sex,
+        ROUND(upper, 4) AS upper
+        FROM data WHERE indicator='$indicator'"""
+    }
```

Comment:
I'm a bit rusty here but I think from when I doing Java a long time ago we would return this as a prepared statement then set the values in it instead of doing the string interpolation.

```
val query = con.prepareStatement(
  """
  SELECT ... from data where indicator = ?
  """)
query.setString(1, indicator)
result = query.executeQuery()
```

I think setting the value in the prepared statement gives protections around sql injection.

Reply:
Ohhh really? I'll read up on it, I'm not very familiar with these things! But definitely sounds very good! Would make this easier for sure!

Reply:
Just read up on it, yh you're definitely right, will refactor to prepared statements when I'm in the office